test.gd

```diff
@@ -3,33 +3,189 @@ extends SceneTree
+    uuid_util.v4_rng(rng)
+    
+  var duration = 1.0 * Time.get_unix_time_from_system() - begin
+
+  print('uuid/sec: %.02f   avg time: %.4fus   total time: %.2fs' % [
+   NUMBER_OF_TESTS / duration,
+   (duration / NUMBER_OF_TESTS) * 1000000,
+   duration
+  ])
+  print('Benchmark done')
+  
+func benchmark_obj():
+  print('Benchmarking ...')
+
+  var begin = Time.get_unix_time_from_system()
+
+  for i in range(NUMBER_OF_TESTS):
+    uuid_util.new().free()  # immediately freeing does not seem to add much overhead
+
+  var duration = 1.0 * Time.get_unix_time_from_system() - begin
+
+  print('uuid/sec: %.02f   avg time: %.4fus   total time: %.2fs' % [
+   NUMBER_OF_TESTS / duration,
+   (duration / NUMBER_OF_TESTS) * 1000000,
+   duration
+  ])
+  print('Benchmark done')
+  
+func benchmark_obj_rng():
+  print('Benchmarking ...')
+
+  var rng = RandomNumberGenerator.new()
+  var begin = Time.get_unix_time_from_system()
 
-  var duration = 1.0 * OS.get_ticks_msec() - begin
+  for i in range(NUMBER_OF_TESTS):
+    uuid_util.new(rng).free()  # immediately freeing does not seem to add much overhead
 
-  print('uuid/sec: %.02f (time: %sms)' % [ NUMBER_OF_TEST / duration, duration])
+  var duration = 1.0 * Time.get_unix_time_from_system() - begin
+
+  print('uuid/sec: %.02f   avg time: %.4fus   total time: %.2fs' % [
+   NUMBER_OF_TESTS / duration,
+   (duration / NUMBER_OF_TESTS) * 1000000,
+   duration
+  ])
+  print('Benchmark done')
+  
+func benchmark_obj_to_dict():
+  print('Setting up benchmark ...')
+  var uuids = []
+  for i in range(NUMBER_OF_OBJECTS):
+    uuids.append(uuid_util.new())
+
+  print('Benchmarking ...')
+  var begin = Time.get_unix_time_from_system()
+
+  for uuid in uuids:
+    uuid.as_dict()
+
+  var duration = 1.0 * Time.get_unix_time_from_system() - begin
+
+  print('uuid/sec: %.02f   avg time: %.4fus   total time: %.2fs' % [
+   NUMBER_OF_TESTS / duration,
+   (duration / NUMBER_OF_TESTS) * 1000000,
+   duration
+  ])
+  print('Cleaning up ...')
+  for uuid in uuids:
+    uuid.free()
+  print('Benchmark done')
+  
+func benchmark_obj_to_str():
+  print('Setting up benchmark ...')
+  var uuids = []
+  for i in range(NUMBER_OF_OBJECTS):
+    uuids.append(uuid_util.new())
+
+  print('Benchmarking ...')
+  var begin = Time.get_unix_time_from_system()
+
+  for uuid in uuids:
+    uuid.as_string()
+
+  var duration = 1.0 * Time.get_unix_time_from_system() - begin
+
+  print('uuid/sec: %.02f   avg time: %.4fus   total time: %.2fs' % [
+   NUMBER_OF_TESTS / duration,
+   (duration / NUMBER_OF_TESTS) * 1000000,
+   duration
+  ])
+  print('Cleaning up ...')
+  for uuid in uuids:
+    uuid.free()
+  print('Benchmark done')
+  
+func benchmark_comp_raw():
+  print('Setting up benchmark ...')
+  var uuids = []
+  for i in range(NUMBER_OF_TESTS*2):
+    uuids.append(uuid_util.v4())
+  print('Benchmarking ...')
+  var begin = Time.get_unix_time_from_system()
+
+  for i in range(NUMBER_OF_TESTS):
+    var uuid1 = uuids[i*2]
```

Comment:
I'd rather see:
```gdscript
for index in range(NUMBER_OF_TESTS * 2):
  if uuids[index] == uuids[index + 1]:
    # TODO: Count number of collisions
    pass
```

Reply:
So you want to overlap comparison pairs?

I.e. 
```
...
uuids[n-2] == uuids[n-1]
uuids[n-1] == uuids[n]
uuids[n]   == uuids[n+1]
uuids[n+1] == uuids[n+2]
...
```

Reply:
Just fixing a potentiel bug in the code I shared before:
```gdscript
for index in range(NUMBER_OF_TESTS * 2 - 1): 
  if uuids[index] == uuids[index + 1]:
    # TODO: Count number of collisions
    pass
```

Regarding the test:
```
...
uuid[0] == uuid[1]
uuid[2] == uuid[3]
uuid[4] == uuid[5]
...
```
But what if `uuid[1] == uuid[2]`, it seems the test misses it.